feel-engine/src/main/scala/org/camunda/feel/FeelEngine.scala

```diff
@@ -175,8 +201,10 @@ class FeelEngine(val functionProvider: FunctionProvider =
 
   def parseExpression(expression: String): Either[Failure, ParsedExpression] =
     parse(FeelParser.parseExpression, expression)
+      .flatMap(validate)
```

Comment:
Question on design: Why do you validate here and not inside the parser?

Reply:
I tried to put it inside the parser but that does not work with the parser library (in a good way). So, it decided to validate after the parsing.